engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTexture.java

```diff
@@ -79,20 +84,22 @@ public SetInputTexture(int textureSlot, int textureId, ResourceUrn materialUrn,
      * The constructor, to be used in the initialise method of a node.
      *
      * Sample use:
-     *      addDesiredStateChange(new SetInputTexture(0, "engine:waterTex", "engine:prog.chunk", "textureWater"));
+     *      addDesiredStateChange(new SetInputTexture(GL_TEXTURE_2D, 0, water.getId(), "engine:prog.chunk", "textureWater"));
      *
      * Instances of this class bind a texture to a texture unit. The integer identifying the texture unit is then
      * passed to a shader program using the material/parameter pair provided on construction. This allow for a
      * texture asset to be used by a shader program as an input.
      *
      * See the source of the process() method for the nitty gritty details.
      *
+     * @param target the attachment point (like TEXTURE_2D) where the texture will be attached to.
```

Comment:
I'd write:

`* @param textureType an opengl constant, can be GL_TEXTURE_2D, GL_TEXTURE_3D and any other texture type listed in https://www.khronos.org/opengl/wiki/Texture#Theory`